proteus/NumericalSolution.py

```diff
@@ -239,26 +239,51 @@ def __init__(self,so,pList,nList,sList,opts,simFlagsList=None):
+                        check_call(gmsh_cmd, shell=True)
+                        logEvent("Done running gmsh; converting to triangle")
+                        MeshTools.msh2triangle(p.domain.geofile)
+
+                    comm.barrier()
                     mesh = MeshTools.TriangularMesh()
-                    mesh.generateFromTriangleFiles(filebase=p.domain.polyfile,base=1)
-                mlMesh = MeshTools.MultilevelTriangularMesh(0,0,0,skipInit=True,
-                                                            nLayersOfOverlap=n.nLayersOfOverlapForParallel,
-                                                            parallelPartitioningType=n.parallelPartitioningType)
-                logEvent("Generating %i-level mesh from coarse Triangle mesh" % (n.nLevels,))
-                mlMesh.generateFromExistingCoarseMesh(mesh,n.nLevels,
-                                                      nLayersOfOverlap=n.nLayersOfOverlapForParallel,
-                                                      parallelPartitioningType=n.parallelPartitioningType)
+                    mlMesh = MeshTools.MultilevelTriangularMesh(0,0,0,skipInit=True,
+                                                                nLayersOfOverlap=n.nLayersOfOverlapForParallel,
+                                                                parallelPartitioningType=n.parallelPartitioningType)
+                    logEvent("Generating %i-level mesh from coarse Triangle mesh" % (n.nLevels,))
+                    logEvent("Generating coarse global mesh from Triangle files")
+                    mesh.generateFromTriangleFiles(filebase=p.domain.geofile,base=1)
+                    logEvent("Generating partitioned %i-level mesh from coarse global Triangle mesh" % (n.nLevels,))
+                    mlMesh.generateFromExistingCoarseMesh(mesh,n.nLevels,
+                                                          nLayersOfOverlap=n.nLayersOfOverlapForParallel,
+                                                          parallelPartitioningType=n.parallelPartitioningType)
+                else:
+                    logEvent("Calling Triangle to generate 2D mesh for"+p.name)
+                    tmesh = TriangleTools.TriangleBaseMesh(baseFlags=n.triangleOptions,
+                                                        nbase=1,
+                                                        verbose=10)
+                    if comm.isMaster() and p.genMesh:
+                        tmesh.readFromPolyFile(p.domain.polyfile)
+                        tmesh.writeToFile(p.domain.polyfile)
+                        logEvent("Converting to Proteus Mesh")
+                        mesh=tmesh.convertToProteusMesh(verbose=1)
+                    comm.barrier()
+                    if not comm.isMaster() or not p.genMesh:
+                        mesh = MeshTools.TriangularMesh()
+                        mesh.generateFromTriangleFiles(filebase=p.domain.polyfile,base=1)
```

Comment:
File "/home/HR/asd/PROTEUS/proteus/proteus/NumericalSolution.py", line 278, in __init__
    mesh.generateFromTriangleFiles(filebase=p.domain.polyfile,base=1)
  File "/home/HR/asd/PROTEUS/proteus/proteus/MeshTools.py", line 3826, in generateFromTriangleFiles
    cmeshTools.generateFromTriangleFiles(self.cmesh,filebase,base)
I get this error, I will post the files that generate it shortly